tests/foreman/api/test_rhcloud_inventory.py

```diff
@@ -90,15 +91,28 @@ def test_rhcloud_inventory_api_e2e(
     assert virtual_host.hostname in hostnames
     assert baremetal_host.hostname in hostnames
     # Verify IP addresses are present in report.
-    ip_addresses = [
-        host['system_profile']['network_interfaces'][0]['ipv4_addresses'][0]
-        for host in json_data['hosts']
-    ]
-    ipv4_addresses = [host['ip_addresses'][0] for host in json_data['hosts']]
-    assert virtual_host.ip_addr in ip_addresses
-    assert baremetal_host.ip_addr in ip_addresses
-    assert virtual_host.ip_addr in ipv4_addresses
-    assert baremetal_host.ip_addr in ipv4_addresses
+    if module_target_sat.network_type == NetworkType.IPV6:
+        # For IPv6 networks, check IPv6 addresses
+        ip_addresses = [
+            host['system_profile']['network_interfaces'][0]['ipv6_addresses'][0]
+            for host in json_data['hosts']
+        ]
+        ipv6_addresses = [host['ip_addresses'][0] for host in json_data['hosts']]
+        assert virtual_host.ip_addr in ip_addresses
+        assert baremetal_host.ip_addr in ip_addresses
+        assert virtual_host.ip_addr in ipv6_addresses
+        assert baremetal_host.ip_addr in ipv6_addresses
+    else:
+        # For IPv4 networks, check IPv4 addresses
+        ip_addresses = [
+            host['system_profile']['network_interfaces'][0]['ipv4_addresses'][0]
+            for host in json_data['hosts']
+        ]
+        ipv4_addresses = [host['ip_addresses'][0] for host in json_data['hosts']]
+        assert virtual_host.ip_addr in ip_addresses
+        assert baremetal_host.ip_addr in ip_addresses
+        assert virtual_host.ip_addr in ipv4_addresses
+        assert baremetal_host.ip_addr in ipv4_addresses
```

Comment:
@ColeHiggins2 Could you please try it this way, I find it more compact. Should work too imo:
```suggestion
		# Dynamic key determination based on the network type
		is_ipv6 = module_target_sat.network_type == NetworkType.IPV6
		key = 'ipv6_addresses' if is_ipv6 else 'ipv4_addresses'
		
		ip_addresses = {
		    host['system_profile']['network_interfaces'][0][key][0]
		    for host in json_data['hosts']
		}
		
		ips = {host['ip_addresses'][0] for host in json_data['hosts']}
		
		for host_ip in (virtual_host.ip_addr, baremetal_host.ip_addr):
		    assert host_ip in ip_addresses
		    assert host_ip in ips
```